File: packages/dotnet/Ygo74.AgentRuntime/Domains/Contracts/ExchangeModels.cs
namespace Ygo74.AgentRuntime.Domains.Contracts;

public sealed record StandardExchangeRequest(
    string RequestId,
    string RouteKey,
    string EndpointType,
    object Input,
    bool Stream = false,
    IDictionary<string, object?>? Metadata = null,
    IDictionary<string, object?>? AuthContext = null
);

public sealed record StandardExchangeResponse(
    string RequestId,
    string Status,
    object? Output = null,
    ErrorEnvelope? Error = null,
    IDictionary<string, object?>? Metadata = null
);

File: packages/java/ygo74-agent-runtime/src/main/java/com/ygo74/agentruntime/domains/contracts/ExchangeModels.java
package com.ygo74.agentruntime.domains.contracts;

import java.util.Map;

public final class ExchangeModels {
    private ExchangeModels() {}

    public record StandardExchangeRequest(
            String requestId,
            String routeKey,
            String endpointType,
            Object input,
            boolean stream,
            Map<String, Object> metadata,
            Map<String, Object> authContext) {
    }

    public record StandardExchangeResponse(
            String requestId,
            String status,
            Object output,
            ErrorEnvelope error,
            Map<String, Object> metadata) {
    }
}
